### ISSUE TITLE
Explicit provider not being used for awsx.ec2.Vpc

### ISSUE THREAD

**Issue by clstokes** [pulumi/pulumi-awsx#314]
Setting an explicit provider does not work for `awsx.ec2.Vpc` and potentially child resources. The ambient provider is still used.

```
$ pulumi config set aws:region does-not-exist
```

```
import * as aws from "@pulumi/aws";
import * as awsx from "@pulumi/awsx";

const east = new aws.Provider("east", { region: "us-east-1" });

const vpc = new awsx.ec2.Vpc("east", {}, {
    providers: {
        "aws": east,
    }
})
```

```
07:49:27 [pulumi-awsx-region-provider] $ pulumi up
Previewing update (dev):

     Type                     Name                             Plan       Info
 +   pulumi:pulumi:Stack      pulumi-awsx-region-provider-dev  create     24 messages
 +   ├─ awsx:x:ec2:Vpc        east                             create
 +   ├─ pulumi:providers:aws  east                             create
     └─ aws:ec2:Vpc           east                                        1 error

Diagnostics:
  pulumi:pulumi:Stack (pulumi-awsx-region-provider-dev):
    Error: invocation of aws:index/getAvailabilityZones:getAvailabilityZones returned an error: Invalid AWS Region: does-not-exist
        at /Users/clstokes/cc/scratch/pulumi-awsx-region-provider/node_modules/@pulumi/pulumi/runtime/invoke.js:74:33
        at Object.onReceiveStatus (/Users/clstokes/cc/scratch/pulumi-awsx-region-provider/node_modules/grpc/src/client_interceptors.js:1205:9)
        at InterceptingListener._callNext (/Users/clstokes/cc/scratch/pulumi-awsx-region-provider/node_modules/grpc/src/client_interceptors.js:
...
```

**Comment by CyrusNajmabadi** [comment]
Easy to fix, but easy bug to make.  I'm going to bring this up at our backcompat discussion tomorrow.  If we're willing to break things, we might want to put on the table the idea of just outright moving to a model where "parent" is required.  It's basically a footgun for us that keeps hitting us over and over again.

**Comment by lukehoban** [comment]
For testing this - I’d suggest we make all tests use 1st class providers and set an invalid `aws:region` for the default provider. That will ensure that we see immediate failures if providers are not being passed through correctly.

**Comment by CyrusNajmabadi** [comment]
> For testing this - I’d suggest we make all tests use 1st class providers and set an invalid aws:region for the default provider. That will ensure that we see immediate failures if providers are not being passed through correctly.

Going to open an issue on this.

**Comment by azkore** [comment]
Could you please recheck that the issue is indeed fixed? I get the same error with 0.18.5 as well as with the package built from master.
